src/lndUtils.ts

```diff
@@ -141,3 +166,146 @@ export const updateRoutingFees = async () => {
+  const diff = escrowInLnd + escrowInMongodb
+
+  baseLogger.info({diff, escrowInLnd, escrowInMongodb, channels}, "escrow recording")
+
+  if (diff > 0) {
+    await MainBook.entry("escrow")
+      .credit(lndAccountingPath, diff, {...metadata})
+      .debit(escrowAccountingPath, diff, {...metadata})
+      .commit()
+  } else if (diff < 0) {
+    await MainBook.entry("escrow")
+      .debit(lndAccountingPath, - diff, {...metadata})
+      .credit(escrowAccountingPath, - diff, {...metadata})
+      .commit()
+  }
+}
+
+
+export const onChannelUpdated = async ({ channel, lnd, stateChange }: { channel: any, lnd: any, stateChange: "opened" | "closed" }) => {
+  baseLogger.info({ channel, stateChange }, `channel update`)
+
+  if (channel.is_partner_initiated) {
+    return
+  }
+
+  if (stateChange === "closed") {
+    // FIXME: need to account for channel closing
+    return
+  }
+  
+  let txid
+
+  if (stateChange === "opened") {
+    ({ transaction_id: txid } = channel)
+  } else if (stateChange === "closed") {
+    ({ close_transaction_id: txid } = channel)
+  }
+  
+  // TODO: dedupe from onchain
+  const { current_block_height } = await getHeight({ lnd })
+  const after = Math.max(0, current_block_height - LOOK_BACK) // this is necessary for tests, otherwise after may be negative
+  const { transactions } = await getChainTransactions({ lnd, after })
+  // end dedupe
+
+  const tx = _.find(transactions, { id: txid })
+
+  if (!tx?.fee) {
+    baseLogger.error({transactions}, "fee doesn't exist")
+    return
+  }
+
+  const fee = tx.fee
+
+  // let tx
+  // try {
+  //   tx = await bitcoindDefaultClient.getRawTransaction(txid, true /* include_watchonly */ )
+  // } catch (err) {
+  //   logger.error({err}, "can't fetch fee for closing tx")
+  // }
+  
+  // TODO: there is no fee currently given by bitcoind for raw transaction
+  // either calculate it from the input, or use an indexer 
+  // const { fee } = tx.fee
+
+  const metadata = { currency: "BTC", txid, type: "fee", pending: false }
+
+  assert(fee > 0)
+
+  await MainBook.entry(`channel ${stateChange} onchain fee`)
+    .debit(lndFeePath, fee, { ...metadata, })
+    .credit(lndAccountingPath, fee, { ...metadata })
+    .commit()
+
+  baseLogger.info({ channel, fee, ...metadata }, `${stateChange} channel fee added to mongodb`)
+}
+
+
+export const getLnds = ({type, active}: {type?: nodeType, active?: boolean} = {}): ILndParamsAuthed[] => {
+  let result = params
+
+  if (!!type) {
+    result = _.filter(result, item => item.type.some(item => item === type))
+  }
+
+  if (!!active) {
+    result = _.filter(result, {active})
+  }
+
+  return result
+}
+
+export const offchainLnds = getLnds({type: "offchain"}) 
+
+// only returning the first one for now
+export const getActiveLnd = () => {
+  const lnds = getLnds({active: true, type: "offchain"})
+  if (lnds.length === 0) {
+    throw Error("no active lnd to send/receive a payment offchain")
+  }
+  // this favor lnd1
+  return lnds[0]
```

Comment:
Have you given more thought to making this function act like a loadbalancer (alternating between lnd1 and lnd2)?

Reply:
yup could be an option. specifically if we get higher and higher load. a random() would be good for that.

but I think for now the main issue is not as much as the load on a single lnd as the need for a failover. 

we can revisit when we have more input with higher load.

Reply:
It might make sense to mock this function as a loadbalancer just for the tests (for now) to make sure the tests work irrespective of which lnd is returned. This might help catch some bugs earlier.